public/js/contribute/pad/render.js

```diff
@@ -939,22 +948,29 @@ function deleteImg(kwargs) {
     } else mosaic.classed('x2', items.size() < 3);
 
     if (editing) {
-      if (page.type === 'private') switchButtons(lang);
+      if (page.type === 'private') await switchButtons(lang);
       else window.sessionStorage.setItem('changed-content', true);
     }
   }
 }
-function addImgs(kwargs) {
+async function addImgs(kwargs) {
   const { data, lang, sibling, container, focus, objectdata } = kwargs;
   const fls = data.filter((d) => d.status === 200);
   // THE CONFIG WITH DATA HERE IS A BIT ANNOYING, BUT IT IS FOR CASES WITH A TEMPLATE, TO MAKE SURE THE VARS SET (e.g. THE INSTRUCTION) ARE MAINTAINED
   if (fls.length === 1) {
-    fls.forEach((f) => {
+    fls.forEach(async (f) => {
```

Comment:
might probably be better to use Promise.all here but I'm not sure if the order is important